### ISSUE TITLE
SOCKS over QUIC

### ISSUE THREAD

**Issue by madhanraj** [quicwg/base-drafts#2518]
Currently, the SOCKS and QUIC don't work straight forward. 
When SOCKS is used, a REQUEST (method, dest port, dest IP) has to be sent during each time of connection establishment. This is straight forward in TCP but in QUIC we establish the connection once with the SOCKS server and multiplex the streams amongst it. 
**Hence, a new method for SOCKS over QUIC is proposed where the REQUEST and RESPONSE are modified according to the QUIC design**

Another important issue with SOCKS is the initial overhead. In our real-time results with Samsung Smartphones, we observed that most the application are short-lived and socks delay impacts the user experience.  Hence we have to decide the packets to be packed one the 0-RTT and also if the tunnel can be pre-established with one-time authentication messages being transferred. 

**We would like to know, if there is similar work going on or shall we submit a fresh RFC including the problem and our approaches**

**Comment by larseggert** [comment]
This is unfortunately outside the charter of the working group, which you can find at https://datatracker.ietf.org/wg/quic/about/. You may of course submit an Internet Draft if you wish and ask the group for feedback, but our current charter prevents us from doing any work in this space.

**Comment by DavidSchinazi** [comment]
Hi @madhanraj, I think your work sounds interesting. I would recommend creating a new internet draft and sharing it on the QUIC mailing list please!

**Comment by LPardue** [comment]
@madhanraj I too think this is interesting and second @DavidSchinazi  recommendation. There are some people thinking about this area and more use cases are welcome, even if we are tangential to the main focus of the group.

I recommend some of the following resources:

https://datatracker.ietf.org/doc/draft-olteanu-intarea-socks-6/
https://tools.ietf.org/html/draft-pauly-quic-datagram-02
https://tools.ietf.org/html/draft-pardue-httpbis-http-network-tunnelling-01

**Comment by madhanraj** [comment]
Thanks .. I missed it as it was labelled closed. We have designed and experimented in our Samsung Lab and find the solution promising from the legacy. Will share the draft and all the results shortly. 
Also why socks and quic ?
If we have to make a proxy solution for all the application (non http too) and if it has to be application agnostic . We have already deployed a similar approach for MPTCP. Hence SOCKS+QUIC would be a interesting use case
